server/src/main/java/io/crate/metadata/settings/SessionSettings.java

```diff
@@ -44,31 +44,55 @@ public class SessionSettings implements Writeable {
     protected SearchPath searchPath;
     protected boolean hashJoinsEnabled;
     protected boolean errorOnUnknownObjectKey;
-
-    public SessionSettings(StreamInput in) throws IOException {
-        this.userName = in.readString();
-        this.searchPath = SearchPath.createSearchPathFrom(in);
-        this.hashJoinsEnabled = in.readBoolean();
-        if (in.getVersion().onOrAfter(Version.V_4_7_0)) {
-            this.errorOnUnknownObjectKey = in.readBoolean();
-        } else {
-            this.errorOnUnknownObjectKey = true;
-        }
-    }
+    protected int memoryLimit;
 
     @VisibleForTesting
     public SessionSettings(String userName, SearchPath searchPath) {
-        this(userName, searchPath, true, true);
+        this(userName, searchPath, true, true, 0);
     }
 
     public SessionSettings(String userName,
                            SearchPath searchPath,
                            boolean hashJoinsEnabled,
-                           boolean errorOnUnknownObjectKey) {
+                           boolean errorOnUnknownObjectKey,
+                           int memoryLimit) {
         this.userName = userName;
         this.searchPath = searchPath;
         this.hashJoinsEnabled = hashJoinsEnabled;
         this.errorOnUnknownObjectKey = errorOnUnknownObjectKey;
+        this.memoryLimit = memoryLimit;
+    }
+
+
+    public SessionSettings(StreamInput in) throws IOException {
```

Comment:
re-ordered these to have the deserialization and serialization logic next to each other. Makes it easier to verify the implementation